dockerfiles/broker.dockerfile

```diff
@@ -11,8 +11,16 @@ RUN curl -L https://foundry.paradigm.xyz | bash && \
     source /root/.bashrc && \
     foundryup
 
-RUN curl -L https://risczero.com/install | bash && \
-    PATH="$PATH:/root/.risc0/bin" rzup install rust 1.85.0
+# Github token can be provided as a secret with the name githubTokenSecret. Useful
+# for shared build environments where Github rate limiting is an issue.
+RUN --mount=type=secret,id=githubTokenSecret,target=/run/secrets/githubTokenSecret \
+    if [ -f /run/secrets/githubTokenSecret ]; then \
+        GITHUB_TOKEN=$(cat /run/secrets/githubTokenSecret) curl -L https://risczero.com/install | bash && \
+        GITHUB_TOKEN=$(cat /run/secrets/githubTokenSecret) PATH="$PATH:/root/.risc0/bin" rzup install rust 1.85.0; \
+    else \
+        curl -L https://risczero.com/install | bash && \
+        PATH="$PATH:/root/.risc0/bin" rzup install rust 1.85.0; \
+    fi
```

Comment:
im not 100% certain about this but i believe you don't need to pass the github token to the dockerfile itself, it just needs to be available in the env where the docker image is being built [[src](https://docs.docker.com/build/cache/backends/gha/#avoid-github-actions-cache-api-throttling)]

Reply:
I tried that initially but was still having issues until I added this. I also don't think docker builds would be repeatable if they could access env variables outside of the container (also seems like that might be a security risk?)

Reply:
yeah looks like i was wrong about this, although the `--build-arg` is available, the docker image ignores it if it's not declared in the image. To test this I wrote this `Dockerfile`:
```
FROM alpine
RUN echo ${GITHUB_TOKEN}
ARG GITHUB_TOKEN
RUN echo ${GITHUB_TOKEN}
```
and this cli command to test it (you'll have to review the build logs):
```
docker build -t test-docker --build-arg GITHUB_TOKEN=test123 . --progress=plain --no-cache
```